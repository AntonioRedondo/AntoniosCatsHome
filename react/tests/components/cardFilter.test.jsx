import React from "react";
import { shallow } from "enzyme";

import CardFilter from "../../src/js/components/cardFilter.jsx";


describe("Given the CardFilter component", () => {
	describe("When the component is mounted", () => {
		it("Should render children", () => {
			const wrapper = shallow(
				<CardFilter
					onChange={ function() {} } />
			);
			
			expect(wrapper.name()).toBe("input");
			expect(wrapper.prop("placeholder")).toBe("Type to filter by name or description");
			expect(wrapper.prop("onChange")).toEqual(jasmine.any(Function));
		});
	});
	
});
