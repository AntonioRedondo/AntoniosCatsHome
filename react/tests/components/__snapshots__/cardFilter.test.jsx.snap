// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CardFilter component When the component is mounted Should render children 1`] = `
<ForwardRef
  onChange={[Function]}
  placeholder="Filter by name or description"
  type="text"
/>
`;
